File: FrontEnd/GitHubSimulatorUI/src/app/shared/search-engine/model/SearchResult.ts
import { Branch } from "src/app/modules/branch/model/Branch";
import { Comment } from "src/app/modules/comment/model/Comment";
import { Issue } from "src/app/modules/issues/model/Issue";
import { Label } from "src/app/modules/labels/model/Label";
import { Milestone } from "src/app/modules/milestones/model/Milestone";
import { Repository } from "src/app/modules/repositories/model/Repository";

export class SearchResult {
  repositories: Repository[] = [];
  branches: Branch[] = [];
  comments: Comment[] = [];
  issues: Issue[] =[];
  labels: Label[] = [];
  milestones: Milestone[] = [];
}

File: BackEnd/GitHubSimulator.Core/Models/Cache/SearchResult.cs
using GitHubSimulator.Core.Models.AggregateRoots;
using GitHubSimulator.Core.Models.Entities;

namespace GitHubSimulator.Core.Models.Cache;

public class SearchResult
{
    public List<Repository> Repositories { get; set; } = new List<Repository>();
    public List<Branch> Branches { get; set; } = new List<Branch>();
    public List<Comment> Comments { get; set; } = new List<Comment>();
    public List<Issue> Issues { get; set; } = new List<Issue>();
    public List<Label> Labels { get; set; } = new List<Label>();
    public List<Milestone> Milestones { get; set; } = new List<Milestone>();
}
